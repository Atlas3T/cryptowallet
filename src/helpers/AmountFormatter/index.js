/**
 * @TODO James Konrad
 * I've mocked some exchange rates below for the amount() function to use
 * Would make sense to put them somewhere in the Vuex store
 */

import numeral from 'numeral';

const exchangeRates = {
  BTC: [
    {
      currency: 'USD',
      rate: 6431.14,
    },
    {
      currency: 'GBP',
      rate: 4977.06,
    },
  ],
  ETH: [
    {
      currency: 'USD',
      rate: 201.76,
    },
    {
      currency: 'GBP',
      rate: 156.14,
    },
  ],
  ATL: [
    {
      currency: 'USD',
      rate: 22222,
    },
    {
      currency: 'GBP',
      rate: 9999,
    },
  ],
};

export default class AmountFormatter {
  constructor(options) {
    this.amount = options.amount;
    this.coin = options.coin;
    this.currency = options.currency;
    this.format = options.format;
    this.withCurrencySymbol = options.withCurrencySymbol;
    this.prependPlusOrMinus = options.prependPlusOrMinus;
  }

  /**
   * Formats a Number and returns a String
   * @return {String}
   */
  getFormatted() {
    let { amount } = this;
    if (this.coin && this.currency) {
      amount = this.coinToCurrency(this.amount, this.coin, this.currency);
    }

    let formatted = `${numeral(Math.abs(amount)).format(this.format)}`;
    if (this.currency) formatted = `${this.currency.symbol}${formatted}`;

    if (this.prependPlusOrMinus) {
      if (amount < 0) return `- ${formatted}`;
      if (amount > 0) return `+ ${formatted}`;
    }

    return formatted;
  }

  /**
   * Converts a coin into a currency
   * @return {Number}
   */
  coinToCurrency() {
    const { rate } = exchangeRates[this.coin].find(item => item.currency === this.currency.code);
    if (rate) return this.amount * rate;
    return this.amount;
  }
}
