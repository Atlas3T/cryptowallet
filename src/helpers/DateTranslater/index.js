import { date as dateUtil } from 'quasar';

/**
 * Returns a translated date/month.
 * @param  {Number} timestamp
 * @param  {String} format
 * @param  {Object} vm Vue component instance
 * @return {String}
 */
export default function dateTranslater(timestamp, format, vm) {
  return dateUtil.formatDate(timestamp, format, {
    dayNames: [
      'sunday',
      'monday',
      'tuesday',
      'wednesday',
      'thursday',
      'friday',
      'saturday',
    ].map((item) => { return vm.$t(item); }),
    monthNames: [
      'january',
      'february',
      'march',
      'april',
      'may',
      'june',
      'july',
      'august',
      'september',
      'october',
      'november',
      'december',
    ].map((item) => { return vm.$t(item); }),
  });
}
