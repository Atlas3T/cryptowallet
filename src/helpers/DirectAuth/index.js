import TorusSdk from '@toruslabs/torus-direct-web-sdk';

async function login() {
  try {
    // AUTH0_DOMAIN: "torus-test.auth0.com"
    // AUTH0_CLIENT_ID: "sqKRBVSdwa4WLkaq419U7Bamlh5vK1H7"
    // GOOGLE_CLIENT_ID: "876733105116-i0hj3s53qiio5k95prpfmj0hp0gmgtor.apps.googleusercontent.com",
    // FACEBOOK_CLIENT_ID: "2554219104599979",
    // AUTH0

    const torusdirectsdk = new TorusSdk({
      baseUrl: `${window.location.origin}/statics/serviceworker`,
      enableLogging: true,
      proxyContractAddress: '0x4023d2a0D330bF11426B12C6144Cfb96B7fa6183', // details for test net
      network: 'ropsten', // details for test net
    });
    await torusdirectsdk.init({ skipSw: false });

    // const loginDetails = await torusdirectsdk.triggerLogin({
    //   typeOfLogin: 'passwordless',
    //   verifier: 'auth0-atlascity',
    //   clientId: 'LgplOAge5k3n17shHwn7ehfzEoiQQHhm',
    //   jwtParams: {
    //     connection: 'sms',
    //     domain: 'https://atlascity.eu.auth0.com',
    //     login_hint: '+447843812196',
    //   },
    // });

    // return loginDetails;
    return true;
  } catch (error) {
    console.error(error, 'caught');
    return false;
  }
}
export default login;
