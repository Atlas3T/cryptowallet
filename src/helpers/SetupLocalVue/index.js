import Vuex from 'vuex';
import { createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';
import VueI18n from 'vue-i18n';
import messages from '@/i18n/';
import axios from 'axios';
import { sync } from 'vuex-router-sync';
import errorHandlerPlugin from '@/boot/ErrorHandler';
import toasterPlugin from '@/boot/Toaster';
import permissionsPlugin from '@/boot/Permissions';
import appInvitationPlugin from '@/boot/AppInvitation';
import smsPlugin from '@/boot/Sms';
import emailPlugin from '@/boot/Email';
import contactsImportPlugin from '@/boot/ContactsImport';
import * as All from 'quasar'

const { Quasar, date } = All;

const components = Object.keys(All).reduce((object, key) => {
  const val = All[key]
  if (val && val.component && val.component.name != null) {
    object[key] = val
  }
  return object
}, {})


const localVue = createLocalVue();

localVue.use(Quasar, { components });
localVue.use(VueI18n);
localVue.use(Vuex);
// localVue.use(VueRouter);
localVue.prototype.$axios = axios;
errorHandlerPlugin({ Vue: localVue });
toasterPlugin({ Vue: localVue });
permissionsPlugin({ Vue: localVue });
appInvitationPlugin({ Vue: localVue });
smsPlugin({ Vue: localVue });
emailPlugin({ Vue: localVue });
contactsImportPlugin({ Vue: localVue });

/**
 * creates a new router and syncs it with the store using vuex-router-sync
 * @param store
 * @returns router
 */
const createRouter = (store = null) => {
  const router = new VueRouter({
    routes: [
      {
        path: '/contact/:id',
      },
    ],
  });

  if (store) { sync(store, router); }
  return router;
};

const i18n = new VueI18n({ messages, locale: 'en-gb' });

export {
  localVue,
  createRouter,
  i18n,
};
