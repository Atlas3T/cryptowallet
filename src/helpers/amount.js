/**
 * @TODO James Konrad
 * mocked data below
 */

import numeral from 'numeral';

const currencySymbols = {
  USD: '$',
  GPB: 'Â£',
};

const exchangeRates = {
  BTC: [
    {
      currency: 'USD',
      rate: 6431.14,
    },
    {
      currency: 'GPB',
      rate: 4977.06,
    },
  ],
  ETH: [
    {
      currency: 'USD',
      rate: 201.76,
    },
    {
      currency: 'GPB',
      rate: 156.14,
    },
  ],
  ATL: [
    {
      currency: 'USD',
      rate: 22222,
    },
    {
      currency: 'GPB',
      rate: 9999,
    },
  ],
};

export default class Amount {
  constructor(options) {
    this.amount = options.amount;
    this.coin = options.coin;
    this.currency = options.currency;
    this.format = options.format;
    this.withCurrencySymbol = options.withCurrencySymbol;
    this.prependPlusOrMinus = options.prependPlusOrMinus;
  }

  /**
   * Formats a Number and returns a String
   * @return {String}
   */
  get formatted() {
    let { amount } = this;
    if (this.coin && this.currency) {
      amount = this.coinToCurrency(this.amount, this.coin, this.currency);
    }

    let formatted = `${numeral(Math.abs(amount)).format(this.format)}`;
    if (this.currency) formatted = `${currencySymbols[this.currency]}${formatted}`;

    if (this.prependPlusOrMinus) {
      if (amount < 0) return `&#45; ${formatted}`;
      if (amount > 0) return `&#43; ${formatted}`;
    }

    return formatted;
  }

  /**
   * Converts a coin into a currency
   * @return {Number}
   */
  coinToCurrency() {
    const { rate } = exchangeRates[this.coin].find(item => item.currency === this.currency);
    return this.amount * rate;
  }
}
