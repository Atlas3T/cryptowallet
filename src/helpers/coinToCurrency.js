const exchangeRates = {
  btc: [
    {
      currency: 'USD',
      value: 6431.14,
    },
    {
      currency: 'GPB',
      value: 4977.06,
    },
  ],
  eth: [
    {
      currency: 'USD',
      value: 201.76,
    },
    {
      currency: 'GPB',
      value: 156.14,
    },
  ],
  atlas: [
    {
      currency: 'USD',
      value: 22222,
    },
    {
      currency: 'GPB',
      value: 9999,
    },
  ],
};

/**
 * Calcuates the value of a coin in a given currency
 * @param  {Number} amount
 * @param  {String} coin
 * @param  {String} currency
 * @return {Number}
 */

export default function coinToCurrency(amount, coin, currency) {
  const { value } = exchangeRates[coin].find(item => item.currency === currency);
  return amount * value;
}
