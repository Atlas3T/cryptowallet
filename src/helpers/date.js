import { date as dateUtil } from 'quasar';

/**
 * Returns formatted and translated date string
 * @param  {Number} timestamp
 * @param  {String} format
 * @param  {Object} vm Vue component instance
 * @return {String}
 */
export default function date(timestamp, format, vm) {
  return dateUtil.formatDate(timestamp, format, {
    dayNames: [
      'sunday',
      'monday',
      'tuesday',
      'wednesday',
      'thursday',
      'friday',
      'saturday',
    ].map(item => vm.$t(item)),
    monthNames: [
      'january',
      'february',
      'march',
      'april',
      'may',
      'june',
      'july',
      'august',
      'september',
      'october',
      'november',
      'december',
    ].map(item => vm.$t(item)),
  });
}
