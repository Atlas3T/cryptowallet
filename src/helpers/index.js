import AmountFormatter from './AmountFormatter';
import dateTranslater from './DateTranslater';
import Contacts from './Contacts';

export {
  AmountFormatter,
  dateTranslater,
  Contacts,
};
