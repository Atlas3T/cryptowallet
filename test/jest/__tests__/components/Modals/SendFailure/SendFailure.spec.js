import { shallowMount, createWrapper } from '@vue/test-utils';
import SendFailure from '@/components/Modals/SendFailure';
import { localVue, i18n } from '@/helpers/SetupLocalVue';
import Vuex from 'vuex';

describe('SendFailure component', () => {
  let store;
  let wrapper;

  beforeEach(() => {
    store = new Vuex.Store();

    wrapper = shallowMount(SendFailure, {
      i18n,
      localVue,
      store,
    });
  });

  it('renders and matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('has sendFailureModalOpened data value set to false when mounted', () => {
    expect(wrapper.vm.sendFailureModalOpened).toBe(false);
  });

  it('updates it\'s data value if \'sendFailureModalOpened\' is emitted', () => {
    wrapper.vm.$root.$emit('sendFailureModalOpened', true);
    expect(wrapper.vm.sendFailureModalOpened).toBe(true);

    wrapper.vm.$root.$emit('sendFailureModalOpened', false);
    expect(wrapper.vm.sendFailureModalOpened).toBe(false);
  });

  it('updates it\'s sendFailureModalOpened data value and navigates a single route back when dismiss() method is called', () => {
    expect(wrapper.vm.sendFailureModalOpened).toBe(false);
  });
});
