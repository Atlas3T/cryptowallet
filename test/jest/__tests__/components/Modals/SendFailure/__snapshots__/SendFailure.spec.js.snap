// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendFailure component renders and matches snapshot 1`] = `
<div>
  <q-dialog-stub
    contentclass="dark-modal"
    maximized="true"
    persistent="true"
    position="standard"
    transitionhide="slide-down"
    transitionshow="slide-up"
  >
    <div
      class="header-section"
    >
      <div
        class="header-back-button-wrapper"
      >
        <q-btn-stub
          align="center"
          class="icon-btn back-arrow-btn"
          flat="true"
          icon="arrow_back"
          ripple="true"
          size="lg"
        />
      </div>
       
      <h1
        class="header-h1"
      >
        
        Error
      
      </h1>
    </div>
     
    <div
      class="modal-layout-wrapper center-modal-content"
    >
      <div
        class="error-content-wrapper"
      >
        <div
          class="error-icon-wrapper"
        >
          <q-icon-stub
            class="error-icon"
            name="error_outline"
          />
        </div>
         
        <div
          class="error-msg-wrapper"
        >
          <h1>
            Error
          </h1>
           
          <h3>
            Failed to broadcast the transaction
          </h3>
        </div>
         
        <div
          class="send"
        >
          <q-btn-stub
            align="center"
            color="blueish"
            label="Close"
            ripple="true"
            size="md"
          />
        </div>
      </div>
    </div>
  </q-dialog-stub>
</div>
`;
